### PR TITLE
fix: split main script logic into helpers

### PR DESCRIPTION
**Pull request by shaftoe**

### CODE REVIEW

**Comment by shaftoe** [comment]
Ref: https://github.com/probot/create-probot-app/issues/315

File: src/create-probot-app.ts [hunk 1]
```diff
@@ -1,334 +1,25 @@
...
+import { makeScaffolding } from "./helpers/filesystem";
+import { printSuccess, red } from "./helpers/write-help";
+
+function main(): void {
+  const program = getProgram();
+
+  getAnswers(program, program.destination).then((answers) => {
+    makeScaffolding(program.destination, answers, program.overwrite)
+      .then(() => {
+        if (program.gitInit) initGit(program.destination);
```
Comment (MaximDevoir) [inline src/create-probot-app.ts:14]: Is running Git initialization asynchronously by design? We can continue running it synchronously to keep the steps separated.


```suggestion
      .then(async () => {
        if (program.gitInit) await initGit(program.destination);
```

Comment (shaftoe) [reply src/create-probot-app.ts:14]: Yes, I forgot to keep it sync, good catch

File: src/create-probot-app.ts [hunk 2]
```diff
@@ -1,334 +1,25 @@
...
-        minimal: true,
-        quotes: "double",
+import { installAndBuild } from "./helpers/run-npm";
+import { makeScaffolding } from "./helpers/filesystem";
+import { printSuccess, red } from "./helpers/write-help";
+
+function main(): void {
+  const program = getProgram();
+
+  getAnswers(program, program.destination).then((answers) => {
```
Comment (MaximDevoir) [inline src/create-probot-app.ts:11]: Prefer `await` on `getAnswers` to *dedent* the following `.then((answers)...` block for easier readability.

```diff
diff --git a/src/create-probot-app.ts b/src/create-probot-app.ts
index 681894b..ed835f0 100644
--- a/src/create-probot-app.ts
+++ b/src/create-probot-app.ts
@@ -5,21 +5,19 @@ import { installAndBuild } from "./helpers/run-npm";
 import { makeScaffolding } from "./helpers/filesystem";
 import { printSuccess, red } from "./helpers/write-help";
 
-function main(): void {
+async function main(): Promise<void> {
   const program = getProgram();
-
-  getAnswers(program, program.destination).then((answers) => {
-    makeScaffolding(program.destination, answers, program.overwrite)
-      .then(() => {
-        if (program.gitInit) initGit(program.destination);
-      })
-      .then(() => installAndBuild(program.destination, answers.toBuild))
-      .then(() => printSuccess(answers.appName, program.destination))
-      .catch((err) => {
-        console.log(red(err));
-        process.exit(1);
-      });
-  });
+  const answers = await getAnswers(program, program.destination)
+  makeScaffolding(program.destination, answers, program.overwrite)
+    .then(async () => {
+      if (program.gitInit) await initGit(program.destination);
+    })
+    .then(() => installAndBuild(program.destination, answers.toBuild))
+    .then(() => printSuccess(answers.appName, program.destination))
+    .catch((err) => {
+      console.log(red(err));
+      process.exit(1);
+    });
 }

```

Comment (shaftoe) [reply src/create-probot-app.ts:11]: Fine for me, feel free to send the patch already (I don't know yet how to send changes proposals directly in the PR but I saw you did before, I'll figure it out)

File: src/create-probot-app.ts [hunk 3]
```diff
@@ -1,334 +1,25 @@
...
+    makeScaffolding(program.destination, answers, program.overwrite)
+      .then(async () => {
+        if (program.gitInit) await initGit(program.destination);
+      })
+      .then(() => installAndBuild(program.destination, answers.toBuild))
+      .then(() => printSuccess(answers.appName, program.destination))
+      .catch((err) => {
+        console.log(red(err));
+        process.exit(1);
```
Comment (MaximDevoir) [inline src/create-probot-app.ts:20]:
```suggestion
  makeScaffolding(program.destination, answers, program.overwrite)
    .then(async () => {
      if (program.gitInit) await initGit(program.destination);
    })
    .then(() => installAndBuild(program.destination, answers.toBuild))
    .then(() => printSuccess(answers.appName, program.destination))
    .catch((err) => {
      console.log(red(err));
      process.exit(1);
```

File: src/create-probot-app.ts (hunk 2 above)
Comment (MaximDevoir) [inline src/create-probot-app.ts:11]:
```suggestion
  const answers = await getAnswers(program, program.destination);
```

File: src/create-probot-app.ts [hunk 4]
```diff
@@ -1,334 +1,25 @@
...
       });
-    }
   });
```
Comment (MaximDevoir) [inline src/create-probot-app.ts:22]:
```suggestion
```

File: src/create-probot-app.ts [hunk 5]
```diff
@@ -1,334 +1,24 @@
...
-    }
-  }
+import { installAndBuild } from "./helpers/run-npm";
+import { makeScaffolding } from "./helpers/filesystem";
+import { printSuccess, red } from "./helpers/write-help";
 
-  return new Promise((resolve, reject) => {
-    const previousDir: string = process.cwd();
-    process.chdir(destination);
+function main(): void {
```
Comment (MaximDevoir) [inline src/create-probot-app.ts:8]: GH doesn't support multi-line suggestions where lines have been removed... yet. This is the last one.


```suggestion
async function main(): Promise<void> {
```

**Review by MaximDevoir — APPROVED** [review]

**Comment by MaximDevoir** [comment]
Suggestions can be found here: https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request

Also, that last part I should've just edited the file without attempting multiline editing of deleted lines where it wasn't possible... but I'm up too late to think that straight.

**Comment by MaximDevoir** [comment]
PR prefix probably should've been `chore` to avoid a release.

**Comment by shaftoe** [comment]
> PR prefix probably should've been `chore` to avoid a release.

I guess is to late now, right? I guess it's time for me to read semantic-release docs too...

**Comment by MaximDevoir** [comment]
That's alright, I didn't catch it either. I think the triggers are the same on most repos from Probot that use semantic release, which can be found here https://github.com/probot/probot/blob/master/CONTRIBUTING.md#merging-the-pull-request--releasing-a-new-version (`fix`, `feat`, and `BREAKING-CHANGE` trigger releases.)

**Comment by gr2m** [comment]
hey I want to start using @all-contributors for probot, I'd like to get better at giving credit to you both for all your help. Also the @all-contributors app is built with Probot and I help maintain them so it's time I start using it myself :D 

@all-contributors please add @MaximDevoir  for code and review

@all-contributors please add @shaftoe for code

**Comment by gr2m** [comment]
lol that's working great :D let me try again

@all-contributors please add @MaximDevoir for code and review

**Comment by gr2m** [comment]
@all-contributors please add @shaftoe for code

**Comment by gr2m** [comment]
@all-contributors please add @shaftoe for code
